crates/lunatic-process/src/message.rs

```diff
@@ -233,19 +259,21 @@ impl Read for DataMessage {
 
 /// A resource ([`WasmProcess`](crate::WasmProcess), [`TcpStream`](tokio::net::TcpStream),
 /// ...) that is attached to a [`DataMessage`].
-pub enum Resource {
+pub enum Resource<T> {
     None,
-    Process(Arc<dyn Process>),
+    Process(Arc<dyn Process<T>>),
+    Module(Arc<WasmtimeCompiledModule<T>>),
     TcpStream(Arc<TcpConnection>),
     UdpSocket(Arc<UdpSocket>),
     TlsStream(Arc<TlsConnection>),
```

Comment:
The `T` generic spreads really everywhere. I assume it's mostly because of Messages/Signals getting a `T`, because they can contain a `Resource<T>`.

Would it be possible just to do `Arc<dyn Any>` for resources and then have the `take_*` functions have a generic, so they try to downcast to the right type and if not just return `None`. The resources vector would be more of a `Vec<Option<Arc<dyn Any>>>`. All of the resources are already wrapped into a `Arc`. This could even simplify future additions to resources, containing more generics.

Reply:
That cleaned things up quite a bit. I've never used dyn Any before really, but came in handy here.

The only downside is I wasn't able to get it working with `fn take_process() -> Arc<dyn Process>` since I can't downcast into `dyn Process`. The `take_process` function though is actually not used anywhere throughout the project, so it doesn't seem to be much of an issue for now.

Reply:
Yeah, that can be removed. Processes are now addressable through the ID and don't need to be put as resources into messages.

This change reduced the added lines by 50%! Looks great! I will merge it as soon as you remove the commented out `take_process` stuff.

Reply:
It looks like it was removed already, my bad. I looked at the wrong diff.

Reply:
I'm not quite sure which part you're referring to? You might have viewed before this commit: https://github.com/lunatic-solutions/lunatic/pull/148/commits/7c9201351f038d8ac330d200ef9eaa3e9a1133b7 ?